crates/fuel-core/src/graphql_api/database.rs

```diff
@@ -156,42 +157,48 @@ impl DatabaseBlocks for ReadView {
 
     fn blocks(
         &self,
-        height: Option<BlockHeight>,
+        height: BlockHeightQuery,
         direction: IterDirection,
     ) -> BoxedIter<'_, StorageResult<CompressedBlock>> {
         // Chain together blocks from the off-chain db and the on-chain db
         // The blocks in off-chain db, if any, are from time before regenesis
 
-        if let Some(height) = height {
+        if let BlockHeightQuery::Specific(height) = height {
             match (height >= self.genesis_height, direction) {
-                (true, IterDirection::Forward) => {
-                    self.on_chain.blocks(Some(height), direction)
-                }
+                (true, IterDirection::Forward) => self
+                    .on_chain
+                    .blocks(BlockHeightQuery::Specific(height), direction),
```

Comment:
This seems like a smell. We are matching on the `BlockHeightQuery` then we reconstruct the `BlockHeightQuery`. I think `blocks` and `old_blocks` can probably just take `Option<BlockHeight>` and we can get rid of the `From` impl into `Option`./F

Reply:
The only issue with this is that `blocks` is a port so it can either take `Option<BlockHeight>` or `BlockHeightQuery` so then this `blocks` function in `database.rs` would have to take a `Option<BlockHeight>`.  What's the difference between reconstructing `BlockHeightQuery` and reconstructing the `Option` like the previous implementation?

Reply:
> What's the difference between reconstructing BlockHeightQuery and reconstructing the Option like the previous implementation?

There's no difference. I think it was a smell in the old code too.

How about
```
        if let BlockHeightQuery::Specific(inner) = height {
            match (inner >= self.genesis_height, direction) {
                (true, IterDirection::Forward) => self
                    .on_chain
                    .blocks(height, direction),
```

It would be nice if we had some type guarantee since the `inner >= self.genesis_height` is kinda redundant.